Automattic/jetpack
# Catch Mapkit API request failures & show error

Fixes https://github.com/Automattic/wp-calypso/issues/74160

## Proposed changes:
This small PR catches when the Mapkit API token retrieval throws an error. It then shows a friendly error message.

This PR contains the changes from [this PR](https://github.com/Automattic/jetpack/pull/29301) as well, since merging would otherwise be hard. It's best to keep off reviewing this Pr, until the other PR is merged.

### Other information:


## Jetpack product discussion

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
1. Apply this PR
2. Add a new post/page
3. Open `projects/plugins/jetpack/extensions/blocks/map/mapkit-utils/index.js` and change the URL on line 114 to something non-existant.
4. Refresh
5. You should see the error message

<img width="858" alt="CleanShot 2023-03-08 at 12 32 05@2x" src="https://user-images.githubusercontent.com/528287/223702616-ac0988dd-e321-41f7-bbd2-d7a3b2b0dd26.png">
